packages/gatsby-source-filesystem/package.json

```diff
@@ -30,10 +30,11 @@
     "gatsby-plugin"
   ],
   "license": "MIT",
+  "main": "index.js",
   "repository": "https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-source-filesystem",
   "scripts": {
     "build": "babel src --out-dir . --ignore __tests__",
-    "prepublish": "cross-env NODE_ENV=production npm run build",
+    "prepare": "cross-env NODE_ENV=production npm run build",
```

Comment:
https://docs.npmjs.com/misc/scripts#deprecation-note

“Since npm@1.1.71, the npm CLI has run the prepublish script for both npm publish and npm install, because it's a convenient way to prepare a package for use (some common use cases are described in the section below). It has also turned out to be, in practice, very confusing. As of npm@4.0.0, a new event has been introduced, prepare, that preserves this existing behavior.”

Reply:
Could you please remove this? It sounds like a reasonable change but it makes it much harder to review your PRs if you're including multiple unrelated changes.

This is also something, that if we make the change, we'd want to do for all packages.

Could you file an issue about making this change so we can discuss it there?